### ISSUE TITLE
test_language_binary_missing test failure when building 0.48.x on Fedora

### ISSUE THREAD

**Issue by jonathanspw** [Textualize/textual#4095]
I'm working on updating the Fedora package to 0.48.1 and I'm running into the following test failure:

```
=================================== FAILURES ===================================
_________________________ test_language_binary_missing _________________________

obj = <module 'textual.document._syntax_aware_document' from '/builddir/build/BUILDROOT/python-textual-0.48.1-1.fc40.x86_64/usr/lib/python3.12/site-packages/textual/document/_syntax_aware_document.py'>
name = 'get_language', ann = 'textual.document._syntax_aware_document'

    def annotated_getattr(obj: object, name: str, ann: str) -> object:
        try:
>           obj = getattr(obj, name)
E           AttributeError: module 'textual.document._syntax_aware_document' has no attribute 'get_language'

/usr/lib/python3.12/site-packages/_pytest/monkeypatch.py:89: AttributeError

The above exception was the direct cause of the following exception:

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f71a6316270>

    @pytest.mark.syntax
    async def test_language_binary_missing(monkeypatch: pytest.MonkeyPatch):
        # mock a failed installation of tree-sitter-language binaries by
        # raising an OSError from get_language
        def raise_oserror(_):
            raise OSError(
                "/path/to/tree_sitter_languages/languages.so: "
                "cannot open shared object file: No such file or directory"
            )
    
>       monkeypatch.setattr(
            "textual.document._syntax_aware_document.get_language", raise_oserror
        )

tests/text_area/test_languages.py:115: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.12/site-packages/_pytest/monkeypatch.py:105: in derive_importpath
    annotated_getattr(target, attr, ann=module)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = <module 'textual.document._syntax_aware_document' from '/builddir/build/BUILDROOT/python-textual-0.48.1-1.fc40.x86_64/usr/lib/python3.12/site-packages/textual/document/_syntax_aware_document.py'>
name = 'get_language', ann = 'textual.document._syntax_aware_document'

    def annotated_getattr(obj: object, name: str, ann: str) -> object:
        try:
            obj = getattr(obj, name)
        except AttributeError as e:
>           raise AttributeError(
                "{!r} object at {} has no attribute {!r}".format(
                    type(obj).__name__, ann, name
                )
            ) from e
E           AttributeError: 'module' object at textual.document._syntax_aware_document has no attribute 'get_language'

/usr/lib/python3.12/site-packages/_pytest/monkeypatch.py:91: AttributeError
=========================== short test summary info ============================
FAILED tests/text_area/test_languages.py::test_language_binary_missing - Attr...
= 1 failed, 2226 passed, 1 skipped, 3 deselected, 4 xfailed in 95.05s (0:01:35) =
```

I'm not terribly sure what to make of this.

**Comment by darrenburns** [comment]
Tests marked with `@pytest.mark.syntax` are only intended to run when the `syntax` extras are installed e.g. `pip install textual[syntax]`. Are they installed in the environment? There was a point where those extra dependencies weren't available on 3.12, but as far as I'm aware that is now resolved.

**Comment by jonathanspw** [comment]
Ah thanks for connecting the dots for me.  We don't have those deps built in Fedora yet so I'll just skip those tests for now.
